src/AzureClient/AzureClient.cs

```diff
@@ -178,38 +183,45 @@ private async Task<ExecutionResult> SubmitOrExecuteJobAsync(IChannel channel, Az
                 return AzureClientError.JobSubmissionFailed.ToExecutionResult();
             }
 
+            // If the command was not %azure.execute, simply return the job status.
             if (!execute)
             {
                 return await GetJobStatusAsync(channel, MostRecentJobId);
             }
 
+            // If the command was %azure.execute, wait for the job to complete and return the job output.
             channel.Stdout($"Waiting up to {submissionContext.ExecutionTimeout} seconds for Azure Quantum job to complete...");
 
-            using (var cts = new System.Threading.CancellationTokenSource(TimeSpan.FromSeconds(submissionContext.ExecutionTimeout)))
+            using (var executionTimeoutTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(submissionContext.ExecutionTimeout)))
+            using (var executionCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(executionTimeoutTokenSource.Token, cancellationToken))
```

Comment:
Since these using statements both end at the end of scope, it may be cleaner to use the new `using var` syntax to remove a level of indentation.
```suggestion
            using var executionTimeoutTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(submissionContext.ExecutionTimeout));
            using var executionCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(executionTimeoutTokenSource.Token, cancellationToken);
```
Following that, can remove the `{` and L197 and the matching `}` at L213.

Reply:
Makes sense!